generate_diagrams.py

```
@@ -0,0 +1,200 @@
+            if isinstance(node, ast.Import):
+                for alias in node.names:
+                    imports.append(alias.name)
+            elif isinstance(node, ast.ImportFrom):
+                if node.module:
+                    imports.append(node.module)
+    except (SyntaxError, FileNotFoundError):
+        pass
+    return imports
+
+
+def generate_import_graph():
```

Ruff (0.12.2)

`generate_import_graph` is too complex (11 > 10)

(C901)